### ISSUE TITLE
The VMM KV-growth attention-mask fix (#1228) has no falsifiable test -- find a model that actually reads the mask, or prove none does

### ISSUE THREAD

**Issue by justinchuby** [justinchuby/onnx-genai#1229]
#1228 (originally #1163) fixed the VMM KV-growth path so it preserves the attention-mask prefix instead of clearing the whole allocation. That change is correct by construction, but **it has no end-to-end test, because I could not find a configuration where the old, wrong behaviour was observable.**

## What was tried

RTX 4060, `qwen05b-q4`, growth forced with:

```
ONNX_GENAI_CUDA_VMM=1 ONNX_GENAI_CUDA_GRAPH=0 ONNX_GENAI_KV_MIN_BUCKET=8
onnx-genai generate <model> "<long prompt>" --backend native --device cuda --max-new-tokens 120 --greedy
```

Three genuine growth events (`old_len=8→64`, `64→128`, `128→256`, at `valid_len=0/64/128`). Output was **character-identical** with the fix and with it reverted.

## Why that is interesting rather than reassuring

`extend_mask(start, end, expose_len)` writes only the incremental `[start, end)` window of `1`s -- it does not rewrite the prefix each step. So a blanket clear on growth genuinely does leave the prefix zeroed. The only consistent explanation for the identical output is that on this model **nothing reads that mask**: past length is carried by another operand (a `total_sequence_length` scalar, or the KV binding's own logical shape).

That is worth knowing on its own. If the mask binding is inert for the models we actually run, we are maintaining, committing, and growing a tensor for nothing on the hot path.

## What would close this

Either:

1. **Find a model whose attention reads the mask** -- a package whose decoder takes `attention_mask` as a real GQA/Attention operand rather than a vestigial input -- and add a regression test that is RED with the memset restored and GREEN with #1228. That is the outcome I would prefer, since it also makes #1228 falsifiable.

2. Or **establish that no supported decode path reads it**, in which case open a follow-up to stop allocating, committing and growing the mask on the VMM path entirely. Do not skip straight to deleting it -- prove the read-side is absent first.

Either answer is a good answer. What is not acceptable is leaving #1228 permanently unfalsifiable.

**Comment by justinchuby** [comment]
It was tested

**Comment by justinchuby** [comment]
Read from working in `native_decode/cuda.rs` while fixing the reservation overflow (#1264) — posting as reasoning, not code, per the ask.

**What the engine does with `bindings[0]` (the mask), unconditionally:**
- It is a *persistent CUDA device binding*, not a per-step owned upload (comments at cuda.rs:1838-1840, 1870-1871, 1891). Every decode entry point calls `extend_mask(...)`/`extend_mask_ragged(...)` to fill it (cuda.rs:989, 1161, 1318, 1626, 1783, 1863, 2006, 2034, 2176, 2424), and it is bound as the graph input named `attention_mask` on every step.
- So the engine **always writes and binds** `bindings[0]`. Whether it is *read* is therefore not decidable in the engine — it reduces entirely to **whether the model's decoder graph feeds `attention_mask` into an Attention/GQA op that actually consumes it.**

**On the models we have locally, nothing observably reads it — consistent with your qwen05b-q4 result:**
- The `tiny-gemma4-vlm-cuda` fixture (and the synthetic test graphs) wire `attention_mask` through a Cast that cancels out, so logits are mask-independent by construction — inert.
- You already showed qwen05b-q4 output was character-identical with and without #1228 across three genuine forced growths. That matches: on that decoder, past length is carried by another operand (the KV binding's logical shape / total-sequence-length), not by `bindings[0]`.

**So my read: on every decode path we can run locally, `bindings[0]` is maintained but not consumed** — the mask prefix #1228 preserves has no observable effect. I could **not** find a local model whose exported decoder genuinely reads `attention_mask` in Attention, and I did not add a RED/GREEN test because there is nothing here to make it go RED.

**What would actually settle it (needs a model we don't have locally):** an exported decoder that (a) contains an `Attention`/`GroupQueryAttention`/`MultiHeadAttention` op with `attention_mask`/`total_sequence_length` wired to the graph's `attention_mask` input, and (b) is run at a sequence length long enough to force a VMM growth so the prefix-preserve path in #1228 is exercised. Many HF ONNX exports fold the causal mask into the graph or drive it from position/total-seq instead, which is likely why neither of us can make the mask matter. If such a model is worth fetching, that is the test to write; absent it, this stays **honestly unresolved**, and the practical implication stands — if no supported decode path reads `bindings[0]`, the mask allocate/commit/grow work on the VMM path (and the overflow hazard in #1266) could be removed entirely.
